### PR TITLE
#137586733 Fixed accounts tests

### PR DESCRIPTION
**Pull request by munala**

### CODE REVIEW

File: hc/accounts/tests/test_check_token.py [hunk 1]
```diff
@@ -8,6 +8,7 @@ def setUp(self):
         super(CheckTokenTestCase, self).setUp()
         self.profile.token = make_password("secret-token")
         self.profile.save()
+        self.client.login(username="alice", password="password")
```
Comment (egichuri) [inline hc/accounts/tests/test_check_token.py]: When this is done in the setup it means the user is logged in before each test runs. Better do this where it's required.

File: hc/accounts/tests/test_check_token.py [hunk 2]
```diff
@@ -21,8 +22,20 @@ def test_it_redirects(self):
...
+        payload = {"email":"alice@example.org", "password":"password"}
+        response = self.client.post("/accounts/login/", payload, content = "application/json")
+        # chcek redirection to chceks
+        self.assertRedirects(response, "/checks/")
+
+    # Login with a bad token and check that it redirects
+    def test_login_with_bad_token(self):
+        payload = {"email":"alice@example.org", "password":"password"}
+        # bad token
+        self.profile.token = 'dcdf'
```
Comment (egichuri) [inline hc/accounts/tests/test_check_token.py]: I don't think you need this. Test using one that's different from the one set in `SetUp`.

File: hc/accounts/tests/test_check_token.py [hunk 3]
```diff
@@ -21,8 +22,20 @@ def test_it_redirects(self):
...
+    # Login and test it redirects already logged in
+    def test_login_and_redirects_already_logged_in(self):
+        payload = {"email":"alice@example.org", "password":"password"}
+        response = self.client.post("/accounts/login/", payload, content = "application/json")
+        # chcek redirection to chceks
+        self.assertRedirects(response, "/checks/")
+
+    # Login with a bad token and check that it redirects
+    def test_login_with_bad_token(self):
+        payload = {"email":"alice@example.org", "password":"password"}
```
Comment (egichuri) [inline hc/accounts/tests/test_check_token.py]: variable `payload` set but not used.

File: hc/accounts/tests/test_check_token.py [hunk 4]
```diff
@@ -21,8 +22,20 @@ def test_it_redirects(self):
         self.profile.refresh_from_db()
         self.assertEqual(self.profile.token, "")
 
-    ### Login and test it redirects already logged in
-
-    ### Login with a bad token and check that it redirects
+    # Login and test it redirects already logged in
```
Comment (egichuri) [inline hc/accounts/tests/test_check_token.py]: Consider using Docstring conventions for comments.

File: hc/accounts/tests/test_check_token.py [hunk 5]
```diff
@@ -21,8 +22,20 @@ def test_it_redirects(self):
...
+        response = self.client.post("/accounts/login/", payload, content = "application/json")
+        # chcek redirection to chceks
+        self.assertRedirects(response, "/checks/")
+
+    # Login with a bad token and check that it redirects
+    def test_login_with_bad_token(self):
+        payload = {"email":"alice@example.org", "password":"password"}
+        # bad token
+        self.profile.token = 'dcdf'
+        response = self.client.post("/accounts/check_token/alice/rgrn/")
```
Comment (egichuri) [inline hc/accounts/tests/test_check_token.py]: (For practice/learning), use `reverse` for this url.

File: hc/accounts/tests/test_profile.py [hunk 6]
```diff
@@ -18,16 +18,24 @@ def test_it_sends_set_password_link(self):
         self.alice.profile.refresh_from_db()
         token = self.alice.profile.token
         ### Assert that the token is set
+        self.assertNotEqual(token, None)
 
         ### Assert that the email was sent and check email content
+        self.assertIn("Here's a link to set a password for your account on healthchecks.io:",
+                      mail.outbox[-1].body)
+        self.assertRedirects(r, "/accounts/set_password_link_sent/")
```
Comment (egichuri) [inline hc/accounts/tests/test_profile.py]: Would be important to use descriptive variable names.
You can also use reverse here.

File: hc/accounts/tests/test_switch_team.py [hunk 7]
```diff
@@ -11,21 +11,29 @@ def test_it_switches(self):
...
     def test_it_checks_team_membership(self):
         self.client.login(username="charlie@example.org", password="password")
 
         url = "/accounts/switch_team/%s/" % self.alice.username
-        r = self.client.get(url)
-        ### Assert the expected error code
+        response = self.client.get(url)
+        # Assert the expected error code
+        # error no access - access forbidden 403
+        assert response.status_code == 403
```
Comment (egichuri) [inline hc/accounts/tests/test_switch_team.py]: Use unittest asserts

**Review by egichuri — CHANGES_REQUESTED** [review]

**Comment by egichuri** [comment]
@lederp23 Please change the base branch for the PR to `develop`

File: hc/accounts/tests/test_check_token.py [hunk 8]
```diff
@@ -8,34 +8,35 @@ def setUp(self):
...
         self.profile.refresh_from_db()
         self.assertEqual(self.profile.token, "")
 
-    # Login and test it redirects already logged in
     def test_login_and_redirects_already_logged_in(self):
+        """ Login and test it redirects already logged in"""
         payload = {"email":"alice@example.org", "password":"password"}
-        response = self.client.post("/accounts/login/", payload, content = "application/json")
-        # chcek redirection to chceks
+        response = self.client.post(
```
Comment (egichuri) [inline hc/accounts/tests/test_check_token.py]: This  response is for login. You're supposed to check the response for the next request when the user is already logged in.

File: hc/accounts/tests/test_check_token.py [hunk 9]
```diff
@@ -8,34 +8,35 @@ def setUp(self):
...
 
-    # Login with a bad token and check that it redirects
     def test_login_with_bad_token(self):
-        payload = {"email":"alice@example.org", "password":"password"}
-        # bad token
-        self.profile.token = 'dcdf'
+        """ Login with a bad token and check that it redirects """
+
+        # test using random token
         response = self.client.post("/accounts/check_token/alice/rgrn/")
```
Comment (egichuri) [inline hc/accounts/tests/test_check_token.py]: Check the [docs](https://docs.djangoproject.com/en/1.10/ref/urlresolvers/#reverse) for `reverse` usage. Of note is if the url accepts arguments.

File: hc/accounts/tests/test_check_token.py (hunk 8 above)
Comment (munala) [reply hc/accounts/tests/test_check_token.py]: What kind of request are we supposed to make after login?

Comment (egichuri) [reply hc/accounts/tests/test_check_token.py]: @lederp23 You can make a get request to the login page and check the response.

Comment (egichuri) [reply hc/accounts/tests/test_check_token.py]: @lederp23 Ignore this. For the test to work we'd need to change the login endpoint which is beyond the scope of this task.

**Review by egichuri — APPROVED** [review]

**Pull request merged**